usaspending_api/disaster/v2/views/award/count.py

```diff
@@ -0,0 +1,36 @@
+from django.db.models import OuterRef, Q, Exists
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.disaster.v2.views.count_base import CountBase
+from usaspending_api.awards.models import FinancialAccountsByAwards
+from usaspending_api.awards.models import Award
+
+
+class AwardCountViewSet(CountBase):
+    """
+    Obtain the count of Agencies related to supplied DEFC filter.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/disaster/award/count.md"
+
+    required_filters = ["def_codes", "award_type_codes"]
+
+    @cache_response()
+    def post(self, request: Request) -> Response:
+        filters = [
+            Q(award=OuterRef("pk")),
+            self.is_in_provided_def_codes(),
+            self.all_closed_defc_submissions,
+            self.is_non_zero_award_cpe(),
+        ]
+        count = (
+            Award.objects.annotate(include=Exists(FinancialAccountsByAwards.objects.filter(*filters).values("pk")))
```

Comment:
I learned on Thursday that for all of the "Award Spending By ___" sections, "All Awards" should return the amounts/counts for all awards in FABA including awards non-linked to File D awards' table. This new understanding will require the endpoint to be implemented differently:
#1: `award_type_codes` needs to be optional (already is)
#2: when `award_type_codes` isn't provided, the logic needs to search for all awards in File C instead of File D
#3: when `award_type_codes` is provided, this current logic can be executed

Reply:
Small comment in API contract to document behavior for API users: https://github.com/fedspendingtransparency/usaspending-api/pull/2568/files#diff-f78d44fe26b3dbca0ed58bde0cc1a853L40-R43

Reply:
Re 1: I noticed that while the filter is optional in TinyShield models, the logic is expecting the key to exist and throws a 500 when the filter is not provided.

Reply:
I observed the same. Fortunately the same if statement to implement the above logic protects from this issue